src/Symfony/Component/DependencyInjection/Argument/ServiceLocatorArgument.php

```diff
@@ -19,4 +21,25 @@
 class ServiceLocatorArgument implements ArgumentInterface
 {
     use ReferenceSetArgumentTrait;
+
+    private $taggedIteratorArgument;
+
+    /**
+     * ServiceLocatorArgument constructor.
+     * @param array|Reference[]|TaggedIteratorArgument $values
+     */
+    public function __construct($values = [])
+    {
+        if ($values instanceof TaggedIteratorArgument) {
+            $this->taggedIteratorArgument = $values;
+            $this->values = [];
+        } else {
+            $this->values = $values;
```

Comment:
you should call setValues here instead, as was done by the trait - to validate the array

Reply:
true